### PR TITLE
Observation text publisher

### PR DESCRIPTION
**Pull request by jonatan-ivanov**
Example output:

```
START - name='sample.operation', contextualName='CALL sampleOperation', error='null', lowCardinalityTags=[a='1'], highCardinalityTags=[time='2022-02-19T00:29:08.937229Z'], map=[class io.micrometer.api.instrument.Timer$Sample='io.micrometer.api.instrument.Timer$Sample@2a18f23c']
 OPEN - name='sample.operation', contextualName='CALL sampleOperation', error='null', lowCardinalityTags=[a='1'], highCardinalityTags=[time='2022-02-19T00:29:08.937229Z'], map=[class io.micrometer.api.instrument.Timer$Sample='io.micrometer.api.instrument.Timer$Sample@2a18f23c']
ERROR - name='sample.operation', contextualName='CALL sampleOperation', error='java.io.IOException: simulated', lowCardinalityTags=[a='1'], highCardinalityTags=[time='2022-02-19T00:29:08.937229Z'], map=[class io.micrometer.api.instrument.Timer$Sample='io.micrometer.api.instrument.Timer$Sample@2a18f23c']
CLOSE - name='sample.operation', contextualName='CALL sampleOperation', error='java.io.IOException: simulated', lowCardinalityTags=[a='1'], highCardinalityTags=[time='2022-02-19T00:29:08.937229Z'], map=[class io.micrometer.api.instrument.Timer$Sample='io.micrometer.api.instrument.Timer$Sample@2a18f23c']
 STOP - name='sample.operation', contextualName='CALL sampleOperation', error='java.io.IOException: simulated', lowCardinalityTags=[a='1', className='CustomContext', localClassName='CustomContext'], highCardinalityTags=[time='2022-02-19T00:29:08.937229Z', userId='55d37607-6db0-4e36-a2f2-e7978906de8a', localUserId='55d37607-6db0-4e36-a2f2-e7978906de8a'], map=[class io.micrometer.api.instrument.Timer$Sample='io.micrometer.api.instrument.Timer$Sample@2a18f23c']

---

sample.operation(TIMER)[a='1', className='CustomContext', error='simulated', localClassName='CustomContext']; count=1.0, total_time=1.036482778 seconds, max=1.036482778 seconds
```

### CODE REVIEW

File: micrometer-api/src/main/java/io/micrometer/api/instrument/observation/Observation.java [hunk 1]
```diff
@@ -397,6 +390,15 @@ public void remove(Object key) {
             return (T) this.map.get(key);
         }
 
+        /**
+         * Removes an entry from the context.
+         *
+         * @param key key by which to remove an entry
+         */
+        public void remove(Object key) {
```
Comment (marcingrzejszczak) [inline micrometer-api/src/main/java/io/micrometer/api/instrument/observation/Observation.java:398]: Maybe we should return the removed object?

**Review by marcingrzejszczak — COMMENTED** [review]
Lgtm, I've added a small remark

**Review by shakuzen — APPROVED** [review]
Looks like a pretty awesome debugging tool to me. Maybe we want to consider some kind of debug option that registers this.

File: micrometer-api/src/main/java/io/micrometer/api/instrument/observation/Observation.java (hunk 1 above)
Comment (jonatan-ivanov) [reply micrometer-api/src/main/java/io/micrometer/api/instrument/observation/Observation.java:398]: 👍🏼 I just moved this method, did not change the signature but we should definitely do this.

**Comment by jonatan-ivanov** [comment]
@shakuzen 👍🏼 I'm going to implement this soon.

**Pull request merged**